src/main/java/com/meilisearch/sdk/TasksHandler.java

```diff
@@ -26,53 +28,93 @@ public class TasksHandler {
     }
 
     /**
-     * Retrieves the task at the specified index uid with the specified task uid
+     * Retrieves the task with the specified task uid
      *
-     * @param indexUid Index identifier to the requested Task
      * @param taskUid Identifier of the requested Task
      * @return Task instance
      * @throws MeilisearchException if client request causes an error
      */
-    Task getTask(String indexUid, int taskUid) throws MeilisearchException {
-        String urlPath = "/indexes/" + indexUid + "/tasks/" + taskUid;
+    Task getTask(int taskUid) throws MeilisearchException {
+        String urlPath = "/tasks/" + taskUid;
         return httpClient.get(urlPath, Task.class);
     }
 
     /**
-     * Retrieves all TasksHandler at the specified index uid
+     * Retrieves tasks from the client
      *
-     * @param indexUid Index identifier to the requested Tasks
      * @return List of task instance
      * @throws MeilisearchException if client request causes an error
      */
-    Results<Task> getTasks(String indexUid) throws MeilisearchException {
+    TasksResults getTasks() throws MeilisearchException {
+        String urlPath = "/tasks";
 
-        Results<Task> result = httpClient.get(urlPath, Results.class, Task.class);
+        TasksResults result = httpClient.get(urlPath, TasksResults.class);
         return result;
     }
 
     /**
-     * Retrieves the task with the specified task uid
+     * Retrieves tasks from the client
      *
-     * @param taskUid Identifier of the requested Task
-     * @return Task instance
+     * @param param accept by the tasks route
+     * @return List of task instance
      * @throws MeilisearchException if client request causes an error
      */
-    Task getTask(int taskUid) throws MeilisearchException {
-        String urlPath = "/tasks/" + taskUid;
-        return httpClient.get(urlPath, Task.class);
+    TasksResults getTasks(TasksQuery param) throws MeilisearchException {
+        URLBuilder urlb = new URLBuilder();
+        urlb.addSubroute("tasks");
+        urlb.addParameter("limit", param.getLimit());
+        urlb.addParameter("from", param.getFrom());
+        urlb.addParameter("status", param.getStatus());
+        urlb.addParameter("type", param.getType());
+        urlb.addParameter("indexUid", param.getIndexUid());
+        String urlQuery = urlb.getURL();
+
+        TasksResults result = httpClient.get(urlQuery, TasksResults.class);
+        return result;
     }
 
     /**
-     * Retrieves tasks from the client
+     * Retrieves all TasksHandler at the specified index uid
```

Comment:
Not sure if I got this... 👀

Reply:
I modify it it's better?

Reply:
What I found strange was the `TasksHandler` part. Maybe you want to `Retrieves all tasks at the specified index uid`